packages/react/slot/src/Slot.tsx

```diff
@@ -13,28 +13,24 @@ const Slot = React.forwardRef<HTMLElement, SlotProps>((props, forwardedRef) => {
   const { children, ...slotProps } = props;
   const childLength = React.Children.count(children);
 
-  if (childLength === 1) {
-    return (
-      <SlotClone {...slotProps} ref={forwardedRef}>
-        {children}
-      </SlotClone>
-    );
-  }
-
-  return (
+  return childLength === 1 ? (
+    <SlotClone {...slotProps} ref={forwardedRef}>
+      {children}
+    </SlotClone>
+  ) : React.Children.toArray(children).some(isSlottable) ? (
```

Comment:
nit: I personally find nested ternaries difficult to read compared to the early return.